storage/raft_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Ben Darnell
+
+package storage
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/storage/engine"
+)
+
+func internalChangeReplicasArgs(newNodeID proto.NodeID, newStoreID proto.StoreID,
+	remove bool, raftID int64, storeID proto.StoreID) (
+	*proto.InternalChangeReplicasRequest, *proto.InternalChangeReplicasResponse) {
+	args := &proto.InternalChangeReplicasRequest{
+		RequestHeader: proto.RequestHeader{
+			RaftID:  raftID,
+			Replica: proto.Replica{StoreID: storeID},
+		},
+		NodeID:  newNodeID,
+		StoreID: newStoreID,
+		Remove:  remove,
+	}
+	reply := &proto.InternalChangeReplicasResponse{}
+	return args, reply
+}
+
+func TestReplicateRange(t *testing.T) {
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	const (
+		rangeID  = 1
+		nodeID1  = 1
+		storeID1 = 1
+		nodeID2  = 2
+		storeID2 = 3
```

Comment:
No particular reason; I was just noticing that we tend to have `nodeID == storeID == 1` in our tests and I wanted to have some cases where `nodeID != storeID`.